### PR TITLE
Modify to_image() to keep true energy axis

### PR DESCRIPTION
**Pull request by AtreyeeS**
**Description**

This pull request modifies `MapDataset.to_image()` to keep the true energy axis as discussed in #2865 . 

**Dear reviewer**

I have kept the models on the imaged dataset. This makes it easier to make the residual images.
Is this ok?

### CODE REVIEW

File: gammapy/datasets/map.py [hunk 1]
```diff
@@ -909,27 +907,64 @@ def to_image(self, spectrum=None, name=None):
...
             background = self.background_model.evaluate() * mask_safe
             background = background.sum_over_axes(keepdims=True)
-            kwargs["models"] = Models(
-                [BackgroundModel(background, datasets_names=[name])]
-            )
+        else:
+            background = None
+
+        if isinstance(self.edisp, EDispMap):
+            kwargs["edisp"] = self.edisp
```
Comment (registerrier) [inline gammapy/datasets/map.py:917]: `edisp` is unchanged?

File: gammapy/datasets/map.py [hunk 2]
```diff
@@ -909,27 +907,64 @@ def to_image(self, spectrum=None, name=None):
...
+        else:
+            kwargs["edisp"] = None
+            if self.exposure is not None:
+                exposure = _map_spectrum_weight(self.exposure, spectrum)
+                kwargs["exposure"] = exposure.sum_over_axes(keepdims=True)
 
-        if self.psf is not None:
             # TODO: implement PSFKernel.to_image()
-            if not isinstance(self.psf, PSFKernel):
+            if isinstance(self.psf, PSFMap):
```
Comment (registerrier) [inline gammapy/datasets/map.py:927]: Normally this should not be applied right?

File: gammapy/datasets/map.py [hunk 3]
```diff
@@ -909,27 +907,64 @@ def to_image(self, spectrum=None, name=None):
...
             else:
-                # assume exposure at center position
                 kwargs["psf"] = None
 
+        kwargs["models"] = self._copy_models(background=background, name=name)
+
         return self.__class__(**kwargs)
 
+    def _copy_models(self, background=None, name=None):
+        """
```
Comment (registerrier) [inline gammapy/datasets/map.py:937]: Here I suppose the problem is more consistency between different `Dataset` transforming methods. 
I probably wouldn't export more than the transformed background model as is the case now. 

We need to define the correct strategy in the long run.

**Comment by AtreyeeS** [comment]
Thanks @registerrier for your comments. I have removed the copy on the models for the present, but this is something I would like to put back in near future.

Perhaps related, I now keep the parameters of the norm and tilt of the background model of the 3D dataset on imaged dataset. Perhaps this is related to model handling as well, but without this, I do not see much use to the `to_image`.

File: gammapy/datasets/map.py [hunk 4]
```diff
@@ -869,14 +868,13 @@ def to_spectrum_dataset(self, on_region, containment_correction=False, name=None
         return SpectrumDataset(**kwargs)
 
     def to_image(self, spectrum=None, name=None):
-        """Create images by summing over the energy axis.
+        """Create images by summing over the reco energy axis.
+        True energy axis is left unchanged to have proper model evaluation.
 
-        Exposure is weighted with an assumed spectrum,
+        In case an EdispMap is not present, true energy axis is also summed;
```
[the remark this answers is not in the document]
Comment (AtreyeeS) [reply gammapy/datasets/map.py]: It seems a default `edisp` is not necessary. I am not totally sure why, but it seems to be created by default at some point.

**Comment by AtreyeeS** [comment]
Closing this for now, opening a new one.

**Pull request closed without merging**